v-next/hardhat-mocha/src/task-action.ts

```diff
@@ -72,6 +72,15 @@ const testWithHardhat: NewTaskActionFunction<TestActionArguments> = async (
 
   const mochaConfig: MochaOptions = { ...hre.config.test.mocha };
 
+  if (hre.config.test.mocha.parallel === true) {
+    // NOTE: We set the global variables as environment variables here because, as of now,
+    // the global options are not automatically passed to the child processes when the parallel
+    // execution is enabled.
+    if (hre.globalOptions.network !== undefined) {
```

Comment:
What if we set all the globalOptions? I think this error would reappear otherwise.

Maybe the confusion comes from the network manager explictly using `process.env`, but every globalOption can be read from the env. See [here](https://github.com/NomicFoundation/hardhat/blob/main/v-next/hardhat/src/internal/core/global-options.ts#L171).

Reply:
Maybe we can have a `getGlobalOptionEnvVariableName` helper, to keep these things in sync. Maybe even a `setGlobalOptionsAsEnvVariables`.